src/libertem/common/buffers.py

```diff
@@ -267,6 +284,50 @@ def get_view_for_tile(self, partition, tile):
         elif self._kind == "single":
             return self._data
 
+    def get_contiguous_view_for_tile(self, partition, tile):
+        '''
+        Make a cached contiguous copy of the view for a single tile
+        if necessary.
+
+        Currently this is only necessary for :code:`kind="sig"` buffers.
+        Use :meth:`flush` to write back the cache.
+
+        Boundary condition: :code:`tile.tile_slice.get(sig_only=True)`
+        does not overlap for different tiles while the cache is active,
+        i.e. the tiles follow LiberTEM slicing for
+        :meth:`libertem.udf.base.UDFTileMixing.process_tile()`.
+
+        Returns
+        view : np.ndarray
+            View into data or contiguous copy if necessary
+
+        '''
+        if self._kind == "sig":
+            sl = tile.tile_slice.get(sig_only=True)
+            key = tuple(map(slice_to_tuple, sl))
+            if key in self._contiguous_cache:
+                return self._contiguous_cache[key]
+            view = self._data[sl]
+            oldshape = view.shape
+            try:
+                # See if the signal dimension can be flattened
+                # https://docs.scipy.org/doc/numpy/reference/generated/numpy.reshape.html
+                view.shape = (-1, )
+                view.shape = oldshape
```

Comment:
Could also check `view.flags.C_CONTIGUOUS` instead - that way we don't have to pay the exception handling cost for every tile

Reply:
Ok, thx! :-)